crates/ruff_python_formatter/src/expression/expr_bool_op.rs

```diff
@@ -51,20 +77,35 @@ impl FormatNodeRule<ExprBoolOp> for FormatExprBoolOp {
                     )?;
                 }
 
-                write!(
-                    f,
-                    [
-                        op.format(),
-                        space(),
-                        in_parentheses_only_group(&value.format())
-                    ]
-                )?;
+                write!(f, [op.format(), space(),])?;
+
+                match value {
```

Comment:
The two matches seem identical. Can we move the logic to a FormatValue { expr } struct that implements Format?

Reply:
Sure!